midend/copyStructures.cpp

```diff
@@ -73,13 +73,14 @@ const IR::Node *DoCopyStructures::postorder(IR::AssignmentStatement *statement)
        FIXME: this is not correct for header unions and should be fixed.
        The fix bellow, commented-out, causes problems elsewhere.
        https://github.com/p4lang/p4c/issues/3842
-    if (ltype->is<IR::Type_HeaderUnion>())
```

Comment:
I am not sure, but it appears that these changes are attempting to fix the issue #3842 mentioned in comments just before this line, and thus those comments can be removed?

Reply:
Well, the current fix just disables union expansion by default to avoid running into #3842. Ideally, we fix the problem in the copyStructures pass somehow by tracking validity.

Reply:
Ah, this PR is not fixing that issue, if I understand the changes correctly.  You are just disabling incorrect transformations.